sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -550,15 +550,33 @@ case class JsonToStructs(
       s"Input schema ${nullableSchema.catalogString} must be a struct, an array or a map.")
   }
 
-  // This converts parsed rows to the desired output by the given schema.
   @transient
-  lazy val converter = nullableSchema match {
-    case _: StructType =>
-      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next() else null
-    case _: ArrayType =>
-      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getArray(0) else null
-    case _: MapType =>
-      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getMap(0) else null
+  private lazy val castRow = nullableSchema match {
+    case _: StructType => (row: InternalRow) => row
+    case _: ArrayType => (row: InternalRow) =>
+      if (row.isNullAt(0)) {
+        new GenericArrayData(Array())
```

Comment:
I think this is the place `from_json` is different from json data source. A data source must produce data as rows, while the `from_json` can return array or map.

I think the previous behavior also makes sense. For array/map, we don't have the corrupted column,  and returning null is reasonable. Actually I prefer null over empty array/map, but we need more discussion about this behavior.

Reply:
I also thought what is better to return here - `null` or empty `Array`/`MapData`. In the case of `StructType` we return `Row` in the `PERMISSIVE` mode. For consistency should we return empty array/map in this mode too?

Maybe we can consider special mode when we can return `null` for the bad record? For now it is easy to do since we use `FailureSafeParser`.

Reply:
> but we need more discussion about this behavior.

@cloud-fan Should I send an email to the dev list or we can discuss this here?

Reply:
I could revert the recent commits and prepare a separate PR for the behaviour change. WDYT?

Reply:
I think it's okay to return `null` for map and array.